project-reports/2023/2023-Q1-Hyperledger-Sawtooth.md

```diff
@@ -0,0 +1,72 @@
+---
+layout: default
+title: 2023 Q1 Hyperledger Sawtooth
+parent: 2023
+grand_parent: Project Updates
+---
+
+# Project Health
+
+Activity primarily focused on integration of Transact into sawtooth-lib, with
+discussions about simplifying the build processes.
+
+# Required Information
+
+- Have you switched from master to main in all your repos? YEs
+- Have you implemented the [Common Repository
+  Structure](../guidelines/repository-structure.md) in all your repos? Yes
+- Has your project implemented these inclusive language changes listed below to
+  your repo? You can optionally [use the DCI Lint
+  tool](https://github.com/petermetz/gh-action-dci-lint#usage) to make this a
+  recurring action on your repo. Yes
+  - master → main
+  - slave → replicas
+  - blacklist → denylist
+  - whitelist → allowlist
+- Have you added an [Inclusive Language
+  Statement](https://wiki.hyperledger.org/display/TSC/Inclusive+Language+Example)
+  to your project's documentation and/or Wiki pages?
+  Yes: [Statement](https://sawtooth.hyperledger.org/community/contributing.html#inclusive-language)
+
+
+# Questions/Issues for the TSC
+
+No new issues
+
+# Releases
+
+- sawtooth-lib v0.8.0
+
+# Overall Activity in the Past Quarter
+
+Initial work on integrating transact into sawtooth-lib and initial release
+(v0.8.0) was done. This is the first step in merging transact into sawtooth-lib
+and eventually deprecating transact.
+
+Various smaller updates went into sawtooth-lib, mostly to update it to the
+latest Rust compiler and language features.
+
+# Current Plans
+
+The primary project focus currently is accessibility by potential future
+contributors. In particular, the current docker-based build and deployment
```

Comment:
Interested in why docker/container based build and deployment is seen as a barrier, and what would you prefer to use?  We tend to think in the reverse, and find using a container based approach much easier.  Not a criticism — I’m trying to understand the differences from a best practices perspective.

Reply:
> Interested in why docker/container based build and deployment is seen as a barrier, and what would you prefer to use? We tend to think in the reverse, and find using a container based approach much easier. Not a criticism — I’m trying to understand the differences from a best practices perspective.

@agunde406
@swcurran 

+1 What I think of when container based builds are mentioned is the VSCode dev containers which 99% of the time is considered by contributors a nice shortcut to setting up build dependencies in a reproducible way. 

Of course there are occasional issues with this too:
1. the git signing setup might need some extra work and 
2. lower-end computers might suffer from resource exhaustion/slowdown due to the containerization overhead.
3. finally, the other issue (that I've seen) is that docker-in-docker doesn't "just work" on the computers with the fruit logo when they have their new in-house ARM based chips ticking in the machine.

Reply:
It is a barrier because its an unnecessary layer obscuring the software. That layering has complicated nearly every development activity.

Beyond that, many of us have moved to M1+ Macs and docker support is horrific for non-native images. So its either spend time retooling all the docker stuff (increasing CI build times with multi-arch images), or rip it all out. In other similar projects (Rust, etc.) where its been ripped out, we've realized that we never should have been using it in the first place.

None of this has to do with whether we should continue to produce docker images for end-users -- maybe that will still be a thing, just at the end of the pipeline, not at every step.

Reply:
@vaporos just to reconfirm, end users will still choose deployment of Sawtooth as a container in container orchestration engines. What is being removed is the `docker` dependency for build and run for developers. Please comment.